add openai agent

isthaison opened the pull request:
### What problem does this PR solve?
add openai agent
_Briefly describe what this PR aims to solve. Include background context that will help reviewers understand the purpose of the PR._

### Type of change

- [ ] Bug Fix (non-breaking change which fixes an issue)
- [x] New Feature (non-breaking change which adds functionality)
- [ ] Documentation Update
- [ ] Refactoring
- [ ] Performance Improvement
- [ ] Other (please describe):

asiroliu commented:
@isthaison 
Could you provide an example python script demonstrating how to use this new feature?

asiroliu commented:
@isthaison 

1. API requests are not returning any results.
![image](https://github.com/user-attachments/assets/773e8b32-3f51-4aff-a00c-e8a0750f5086)
![image](https://github.com/user-attachments/assets/85a590bb-36dc-4d89-bfa4-60898a74e11d)

2. The documentation doesn't match the actual API behavior and needs correction.
![image](https://github.com/user-attachments/assets/dc042491-9e8b-42c0-ad1b-dda67f4a3656)

isthaison commented:
Hi  @asiroliu  first call to get id _session
subsequent calls pass additional id

isthaison commented:
Hi @asiroliu , please check again

asiroliu commented:
@isthaison 
Hi, the current reproduction and testing process is too cumbersome - it's consuming too much time to understand how to use your API. It would be really helpful if you could provide a brief 'manual' describing all the details for using your API.

isthaison commented:
```python 
import requests
import json
stream = True

payload = {
    "model": "model",
    "messages": [
        {"role": "user", "content": "initial message"},
    ],
    "stream": stream
}
print(f"Initial payload: {payload}")
def send_completion_request(payload):
    url = "http://localhost:9222/api/v1/agents_openai/ed1625fa062011f0a718be1ad1ab1f02/chat/completions"
  
    headers = {
        'key': 'itsmart',
        'Authorization': "Bearer ragflow-FmN2E0NTVlZTc1ODExZWZhNGZiMDI0Mm",
        'Content-Type': 'application/json',
    }

    response = requests.post(url, headers=headers, json=payload, stream=True)
    
    if response.status_code == 200:
        print("Request successful. Response:")
        response_id = None
        for line in response.iter_lines():
            if line:
                decoded_line = line.decode('utf-8')
                print(decoded_line)
                try:
                    data = json.loads(decoded_line[6:])  # Skip "data: " prefix
                    print("=====data: {}=======\n".format(data))
                    if "id" in data:
                        response_id = data["id"]
                    if "choices" in data and len(data["choices"]) > 0:
                        respchoice = data["choices"][0]["message"]
                except json.JSONDecodeError:
                    continue
        
        if response_id:
            print(f"Extracted ID: {response_id}")
            # Use the extracted ID for further requests
            return [response_id, respchoice]
    else:
        print(f"Request failed with status code {response.status_code}.")
        print(response.text)
        return None

# Call the function and store the extracted ID
[response_id, respchoice] = send_completion_request(payload)
print(f"Response ID: {response_id}")
print(f"Response choice: {respchoice}")


payload["id"] = response_id  # Update the payload with the new ID
payload["messages"].append(respchoice)  # Add a new message to the payload
payload["messages"].append({"role": "user", "content": "have a nice day"})  # Add a new message to the payload
print(f"Updated payload: {payload}")
# Call the function again with the updated payload
[response_id, respchoice] = send_completion_request(payload)

print(f"respchoice: {respchoice}")
``` 
=====
print 
```
Initial payload: {'model': 'model', 'messages': [{'role': 'user', 'content': 'initial message'}], 'stream': True}
Request successful. Response:
data: {"id": "8fececb40de211f0a97cba4315475c03", "object": "chat.completion.chunk", "created": null, "model": "ed1625fa062011f0a718be1ad1ab1f02", "param": null, "usage": {"prompt_tokens": 2, "completion_tokens": 15, "total_tokens": 17, "completion_tokens_details": {"reasoning_tokens": 0, "accepted_prediction_tokens": 0, "rejected_prediction_tokens": 0}}, "choices": [{"message": {"role": "assistant", "content": "Hi! I'm your smart assistant. What can I do for you?"}, "logprobs": null, "finish_reason": "stop", "index": 0}]}
=====data: {'id': '8fececb40de211f0a97cba4315475c03', 'object': 'chat.completion.chunk', 'created': None, 'model': 'ed1625fa062011f0a718be1ad1ab1f02', 'param': None, 'usage': {'prompt_tokens': 2, 'completion_tokens': 15, 'total_tokens': 17, 'completion_tokens_details': {'reasoning_tokens': 0, 'accepted_prediction_tokens': 0, 'rejected_prediction_tokens': 0}}, 'choices': [{'message': {'role': 'assistant', 'content': "Hi! I'm your smart assistant. What can I do for you?"}, 'logprobs': None, 'finish_reason': 'stop', 'index': 0}]}=======

data: [DONE]
Extracted ID: 8fececb40de211f0a97cba4315475c03
Response ID: 8fececb40de211f0a97cba4315475c03
Response choice: {'role': 'assistant', 'content': "Hi! I'm your smart assistant. What can I do for you?"}
Updated payload: {'model': 'model', 'messages': [{'role': 'user', 'content': 'initial message'}, {'role': 'assistant', 'content': "Hi! I'm your smart assistant. What can I do for you?"}, {'role': 'user', 'content': 'have a nice day'}], 'stream': True, 'id': '8fececb40de211f0a97cba4315475c03'}
Request successful. Response:
data: {"id": "8fececb40de211f0a97cba4315475c03", "object": "chat.completion.chunk", "created": null, "model": "ed1625fa062011f0a718be1ad1ab1f02", "param": null, "usage": {"prompt_tokens": 4, "completion_tokens": 13, "total_tokens": 17, "completion_tokens_details": {"reasoning_tokens": 0, "accepted_prediction_tokens": 0, "rejected_prediction_tokens": 0}}, "choices": [{"message": {"role": "assistant", "content": "*'Categorize_0'* is running...🕞"}, "logprobs": null, "finish_reason": null, "index": 0}]}
=====data: {'id': '8fececb40de211f0a97cba4315475c03', 'object': 'chat.completion.chunk', 'created': None, 'model': 'ed1625fa062011f0a718be1ad1ab1f02', 'param': None, 'usage': {'prompt_tokens': 4, 'completion_tokens': 13, 'total_tokens': 17, 'completion_tokens_details': {'reasoning_tokens': 0, 'accepted_prediction_tokens': 0, 'rejected_prediction_tokens': 0}}, 'choices': [{'message': {'role': 'assistant', 'content': "*'Categorize_0'* is running...🕞"}, 'logprobs': None, 'finish_reason': None, 'index': 0}]}=======       

data: {"id": "8fececb40de211f0a97cba4315475c03", "object": "chat.completion.chunk", "created": null, "model": "ed1625fa062011f0a718be1ad1ab1f02", "param": null, "usage": {"prompt_tokens": 4, "completion_tokens": 24, "total_tokens": 28, "completion_tokens_details": {"reasoning_tokens": 0, "accepted_prediction_tokens": 0, "rejected_prediction_tokens": 0}}, "choices": [{"message": {"role": "assistant", "content": "*'Generate_0'* is running...🕞"}, "logprobs": null, "finish_reason": null, "index": 0}]}
=====data: {'id': '8fececb40de211f0a97cba4315475c03', 'object': 'chat.completion.chunk', 'created': None, 'model': 'ed1625fa062011f0a718be1ad1ab1f02', 'param': None, 'usage': {'prompt_tokens': 4, 'completion_tokens': 24, 'total_tokens': 28, 'completion_tokens_details': {'reasoning_tokens': 0, 'accepted_prediction_tokens': 0, 'rejected_prediction_tokens': 0}}, 'choices': [{'message': {'role': 'assistant', 'content': "*'Generate_0'* is running...🕞"}, 'logprobs': None, 'finish_reason': None, 'index': 0}]}=======

data: {"id": "8fececb40de211f0a97cba4315475c03", "object": "chat.completion.chunk", "created": null, "model": "ed1625fa062011f0a718be1ad1ab1f02", "param": null, "usage": {"prompt_tokens": 4, "completion_tokens": 24, "total_tokens": 28, "completion_tokens_details": {"reasoning_tokens": 0, "accepted_prediction_tokens": 0, "rejected_prediction_tokens": 0}}, "choices": [{"message": {"role": "assistant", "content": ""}, "logprobs": null, "finish_reason": "stop", "index": 0}]}
=====data: {'id': '8fececb40de211f0a97cba4315475c03', 'object': 'chat.completion.chunk', 'created': None, 'model': 'ed1625fa062011f0a718be1ad1ab1f02', 'param': None, 'usage': {'prompt_tokens': 4, 'completion_tokens': 24, 'total_tokens': 28, 'completion_tokens_details': {'reasoning_tokens': 0, 'accepted_prediction_tokens': 0, 'rejected_prediction_tokens': 0}}, 'choices': [{'message': {'role': 'assistant', 'content': ''}, 'logprobs': None, 'finish_reason': 'stop', 'index': 0}]}=======

data: {"id": "8fececb40de211f0a97cba4315475c03", "object": "chat.completion.chunk", "created": null, "model": "ed1625fa062011f0a718be1ad1ab1f02", "param": null, "usage": {"prompt_tokens": 4, "completion_tokens": 25, "total_tokens": 29, "completion_tokens_details": {"reasoning_tokens": 0, "accepted_prediction_tokens": 0, "rejected_prediction_tokens": 0}}, "choices": [{"message": {"role": "assistant", "content": "You"}, "logprobs": null, "finish_reason": "stop", "index": 0}]}
=====data: {'id': '8fececb40de211f0a97cba4315475c03', 'object': 'chat.completion.chunk', 'created': None, 'model': 'ed1625fa062011f0a718be1ad1ab1f02', 'param': None, 'usage': {'prompt_tokens': 4, 'completion_tokens': 25, 'total_tokens': 29, 'completion_tokens_details': {'reasoning_tokens': 0, 'accepted_prediction_tokens': 0, 'rejected_prediction_tokens': 0}}, 'choices': [{'message': {'role': 'assistant', 'content': 'You'}, 'logprobs': None, 'finish_reason': 'stop', 'index': 0}]}=======

data: {"id": "8fececb40de211f0a97cba4315475c03", "object": "chat.completion.chunk", "created": null, "model": "ed1625fa062011f0a718be1ad1ab1f02", "param": null, "usage": {"prompt_tokens": 4, "completion_tokens": 27, "total_tokens": 31, "completion_tokens_details": {"reasoning_tokens": 0, "accepted_prediction_tokens": 0, "rejected_prediction_tokens": 0}}, "choices": [{"message": {"role": "assistant", "content": "You too"}, "logprobs": null, "finish_reason": "stop", "index": 0}]}
=====data: {'id': '8fececb40de211f0a97cba4315475c03', 'object': 'chat.completion.chunk', 'created': None, 'model': 'ed1625fa062011f0a718be1ad1ab1f02', 'param': None, 'usage': {'prompt_tokens': 4, 'completion_tokens': 27, 'total_tokens': 31, 'completion_tokens_details': {'reasoning_tokens': 0, 'accepted_prediction_tokens': 0, 'rejected_prediction_tokens': 0}}, 'choices': [{'message': {'role': 'assistant', 'content': 'You too'}, 'logprobs': None, 'finish_reason': 'stop', 'index': 0}]}=======

data: {"id": "8fececb40de211f0a97cba4315475c03", "object": "chat.completion.chunk", "created": null, "model": "ed1625fa062011f0a718be1ad1ab1f02", "param": null, "usage": {"prompt_tokens": 4, "completion_tokens": 30, "total_tokens": 34, "completion_tokens_details": {"reasoning_tokens": 0, "accepted_prediction_tokens": 0, "rejected_prediction_tokens": 0}}, "choices": [{"message": {"role": "assistant", "content": "You too."}, "logprobs": null, "finish_reason": "stop", "index": 0}]}
=====data: {'id': '8fececb40de211f0a97cba4315475c03', 'object': 'chat.completion.chunk', 'created': None, 'model': 'ed1625fa062011f0a718be1ad1ab1f02', 'param': None, 'usage': {'prompt_tokens': 4, 'completion_tokens': 30, 'total_tokens': 34, 'completion_tokens_details': {'reasoning_tokens': 0, 'accepted_prediction_tokens': 0, 'rejected_prediction_tokens': 0}}, 'choices': [{'message': {'role': 'assistant', 'content': 'You too.'}, 'logprobs': None, 'finish_reason': 'stop', 'index': 0}]}=======

data: {"id": "8fececb40de211f0a97cba4315475c03", "object": "chat.completion.chunk", "created": null, "model": "ed1625fa062011f0a718be1ad1ab1f02", "param": null, "usage": {"prompt_tokens": 4, "completion_tokens": 33, "total_tokens": 37, "completion_tokens_details": {"reasoning_tokens": 0, "accepted_prediction_tokens": 0, "rejected_prediction_tokens": 0}}, "choices": [{"message": {"role": "assistant", "content": "You too."}, "logprobs": null, "finish_reason": "stop", "index": 0}]}
=====data: {'id': '8fececb40de211f0a97cba4315475c03', 'object': 'chat.completion.chunk', 'created': None, 'model': 'ed1625fa062011f0a718be1ad1ab1f02', 'param': None, 'usage': {'prompt_tokens': 4, 'completion_tokens': 33, 'total_tokens': 37, 'completion_tokens_details': {'reasoning_tokens': 0, 'accepted_prediction_tokens': 0, 'rejected_prediction_tokens': 0}}, 'choices': [{'message': {'role': 'assistant', 'content': 'You too.'}, 'logprobs': None, 'finish_reason': 'stop', 'index': 0}]}=======

data: [DONE]
Extracted ID: 8fececb40de211f0a97cba4315475c03
respchoice: {'role': 'assistant', 'content': 'You too.'}
# image
![image](https://github.com/user-attachments/assets/15c413c3-33a6-4ec8-ae3c-437579b34997)

# Target: integration mattermost

isthaison commented:
Hi @asiroliu   , I can use the example python script  under

asiroliu commented:
@isthaison @KevinHuSh 
LGTM

I can verify this using the method you provided.
![image](https://github.com/user-attachments/assets/64dc7382-a8c2-41f0-950b-beaa8980a4c7)
